### PR TITLE
Updating to purescript 0.7.0

### PR DESCRIPTION
**Pull request by SimonRichardson**
Syntax changes
- Strings lastIndexOf now returns a maybe, so no need to do the let if
   checks.
- Moved to glup, to fall in line with other purescript repos.
- Removed the compiled example (I can put it back if required, but we should use a different folder name as we're using this folder for gulp).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
+  "name"    : "purescript-pathy",
   "private": true,
-  "dependencies": {
-    "grunt": "~0.4.5",
-    "grunt-purescript": "~0.6.0",
-    "grunt-contrib-clean": "~0.6.0",
-    "grunt-jsvalidate": "~0.2.2"
+  "devDependencies": {
+    "ioredis": "1.1.x",
```
Comment (garyb) [inline package.json]: Redis? :smile:

File: bower.json [hunk 2]
```diff
@@ -19,18 +19,19 @@
...
+    "purescript-maybe":                 "^0.3.0",
+    "purescript-strings":               "^0.5.0",
+    "purescript-eff":                   "^0.1.0",
+    "purescript-lists":                 "^0.7.0",
+    "purescript-profunctor":            "^0.3.0",
+    "purescript-control":               "^0.3.0",
+    "purescript-console":               "^0.1.0",
+    "purescript-foldable-traversable":  "^0.4.0",
+    "purescript-transformers":          "^0.6.0"
   }
```
Comment (garyb) [inline bower.json:28]: This is just being picky, but can you reduce this to:

``` json
  "dependencies": {
    "purescript-console":      "^0.1.0",
    "purescript-exceptions":   "^0.3.0",
    "purescript-lists":        "^0.7.0",
    "purescript-profunctor":   "^0.3.0",
    "purescript-strings":      "^0.5.2",
    "purescript-transformers": "^0.6.1"
  }
```

That will pull everything in the above, and doesn't duplicate the `purescript-control` requirement :wink:

File: package.json (hunk 1 above)
Comment (SimonRichardson) [inline package.json]: D'ho

**Comment by jdegoes** [comment]
:+1: Thanks! You using this anywhere or just helping out the OSS community? :smiley:

**Comment by SimonRichardson** [comment]
Learning Purescript essentially, I blame you @jdegoes for creating stax! We do experiment and prototype somethings with Purescript but nothing live atm.

**Comment by jdegoes** [comment]
Ah yes, Stax, the ancient path to FP in haXe. :smiling_imp: I hope I can get you hooked on PureScript.... :smiley:

**Comment by SimonRichardson** [comment]
I seem to be coming back, so must be scratching an inch :)

**Pull request merged**